_drafts/restic-foundation1-cdc.md

```diff
@@ -0,0 +1,264 @@
+
+{% highlight console %}
+$ restic backup testdata
+using parent snapshot 7452bd17
+scan [/home/fd0/tmp/testdata]
+scanned 1 directories, 1 files in 0:00
+[0:00] 100.00%  0B/s  100.000 MiB / 100.000 MiB  0 / 2 items  ... ETA 0:00
+duration: 0:00, 20478.98MiB/s
+snapshot 0b870550 saved
+{% endhighlight %}
+
+Again you can see that we've instructed restic to backup 100MiB of data, but in
+this case restic was much faster and finished the job in less than a second.
+
+Looking at the repository size you can probably already guess that it is still
+about 100MiB, since we didn't really add any new data:
+
+{% highlight console %}
+$ du -sh $RESTIC_REPOSITORY
+101M	/tmp/restic-test-repository
+{% endhighlight %}
+
+When we make a copy of the file `file.raw` and backup the same repository again,
+restic recognises that all data is already known and the repository size does
+not grow at all, although the directory `testdata` now contains 200MiB of data:
+
+{% highlight console %}
+$ cp testdata/file.raw testdata/file2.raw
+
+$ du -sh testdata
+200M	testdata
+
+$ restic backup testdata
+using parent snapshot 0b870550
+scan [/home/fd0/tmp/testdata]
+scanned 1 directories, 2 files in 0:00
+[0:01] 100.00%  163.617 MiB/s  200.000 MiB / 200.000 MiB  1 / 3 ... ETA 0:00
+duration: 0:01, 129.06MiB/s
+snapshot ab8e0047 saved
+
+$ du -sh $RESTIC_REPOSITORY
+101M	/tmp/restic-test-repository
+{% endhighlight %}
+
+Now for the final demonstration we'll create a new file `file3.raw` which is a
+nasty combination of the 100MiB we've initially saved in `file.raw` so that
+`testdata` now contains about 400MiB:
+
+{% highlight console %}
+$ (echo foo; cat testdata/file.raw; echo bar; cat testdata/file.raw; echo baz) \
+  > testdata/file3.raw
+
+$ du -sh testdata
+401M	testdata
+{% endhighlight %}
+
+We'll create a new backup of the directory with restic and observe that the
+repository has grown by about 10MiB:
+
+{% highlight console %}
+$ restic backup testdata
+using parent snapshot ab8e0047
+scan [/home/fd0/tmp/testdata]
+scanned 1 directories, 3 files in 0:00
+[0:03] 100.00%  127.638 MiB/s  400.000 MiB / 400.000 MiB  2 / 4 ... ETA 0:00
+duration: 0:03, 123.73MiB/s
+snapshot a8897ae3 saved
+
+$ du -sh $RESTIC_REPOSITORY
+111M	/tmp/restic-test-repository
+{% endhighlight %}
+
+This is expected because we've created a few new chunks when creating
+`file3.raw`, e.g. the first chunk will be saved again because a few bytes (the
+string `foo\n`) has been added. But restic managed the challenge quite well.
+
+### Conclusion
+
+Content Defined Chunking is a clever idea to split large amounts of data (e.g.
+large files) into small chunks, while being able to recognize the same chunks
+again when shifted or (slightly) modified.
+
+This enables restic to de-duplicate chunks so that each chunk of data is only
+saved in (and transmitted to) the backup location once. This enables not only
+*inter-file* de-duplication, but also the more relevant *inter-backup*
+de-duplication.
+
```

Comment:
In general, one could also add a short note (maybe at the beginning) that points out that a file can change name, directory location on the same machine or even be moved to another machine (that gets backed up into same repo) without harming the de-duplication. It's rather clear from the way CDC works, but other tools (like rsync) have an issue with that because they work based on the name/path.